### PR TITLE
Create hotfix PR template

### PR DESCRIPTION
**Pull request by ColeKettler**
## Overview

Sets up a PR template to document the steps for creating a hotfix for the example site. The example site is not versioned with the library and will usually be released as part of regular library releases, but in cases like #27 it may be necessary to release a version of the site without creating a release for the library.

The hotfix instructions are a little looser than the release instructions. My intent is to strike a good balance between flexibility and having a simple checklist to follow.

## Testing Instructions

- Review and critique hotfix workflow.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE/HOTFIX.md
Comment (ColeKettler) [inline .github/PULL_REQUEST_TEMPLATE/HOTFIX.md:2]: The example site isn't semantically versioned and changes made only to it will not result in a version bump for the library or a corresponding version tag. Using the current date seems like a Good Enough solution for distinguishing hotfixes from each other.

File: .github/PULL_REQUEST_TEMPLATE/HOTFIX.md
Comment (ColeKettler) [inline .github/PULL_REQUEST_TEMPLATE/HOTFIX.md:19]: This lets us test and verify the hotfix on a Netlify deploy preview before merging it, without the overhead of an issue or requiring us to merge unreleased library code from `develop` into `main`.

File: .github/PULL_REQUEST_TEMPLATE/HOTFIX.md
Comment (ColeKettler) [inline .github/PULL_REQUEST_TEMPLATE/HOTFIX.md:22]: Prevents triggering any tag-based workflows and keeps the tag history clean.

File: README.md
Comment (ColeKettler) [inline README.md:435]: Added this in here as a placeholder prior to a `CONTRIBUTING` file being created. I also wanted to mention our preferred version of the `gitflow` CLI. There was some confusion as to where the commands in the release issue template were coming from.

File: .github/workflows/release.yml
Comment (ColeKettler) [inline .github/workflows/release.yml:6]: This is a guardrail against a non-version tag being accidentally pushed and published to npm.

**Comment by ColeKettler** [comment]
Going to wait for #27 to be merged to test out this workflow before submitting this for review.

**Comment by ColeKettler** [comment]
@rbreslow This workflow was exercised in #27.

File: .github/PULL_REQUEST_TEMPLATE/HOTFIX.md
Comment (ColeKettler) [inline .github/PULL_REQUEST_TEMPLATE/HOTFIX.md:25]: #27 was squashed with the `-S` option in an attempt to get it more like our typical feature branch merges, but in hindsight I don't love the default commit message formatting and think it would have been more valuable to preserve the individual commits. I removed the `-S` option from the `hotfix finish` command as a result.

File: .github/workflows/release.yml
Comment (rbreslow) [reply .github/workflows/release.yml:6]: I don't think this will work as it also matches `1970-01-01` as an example. You may want something like this:

```suggestion
      - "^[0-9]+\.[0-9]+\.[0-9]+$"
```

`^` matches the beginning of the string, then we escape `.` to match the character literal, and `$` matches the end of the string. 

Without the `^` and `$`, something like `1.1.1.1.1.1.1.1.1.1.1` would match multiple capture groups.

**Review by rbreslow — COMMENTED** [review]
This looks OK. I left a comment about the tag RegEx.

 It doesn't feel perfect—I'm worried it may confuse folks without the example site context. Like, `hotfix/` isn't a hotfix for react-showtime, it's a hotfix for something else. At the same time, I'm not sure of an alternative or how much value it'd generate to research one.

File: .github/workflows/release.yml
Comment (ColeKettler) [reply .github/workflows/release.yml:6]: The syntax for filtering tags and branches looks regexy, but it's actually got its [own filtering syntax](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet)! An example semver pattern is provided in that cheatsheet.

Apparently regex filters isn't something that is supported in workflows. [This thread](https://github.community/t/using-regex-for-filtering/16427/2) is a little old, but I haven't found anything more recent that indicates to the contrary.

**Comment by ColeKettler** [comment]
I agree, I don't think it's a perfect match for Git Flow semantics. I don't think there's even much of a concept for hotfixing a semver library like this, it'd just be a hasty patch put through the same release workflow. We're kinda coopting the hotfix concept here, but I wasn't able to think of anything less complicated or clearer.

**Review by rbreslow — APPROVED** [review]
👍

**Pull request merged**
